common/src/main/java/io/druid/data/SearchableVersionedDataFinder.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.data;
+
+import java.util.regex.Pattern;
+
+/**
+ * A class which implements SearchableVersionedDataFinder can find a DataDescriptor which describes the most up to
+ * date version of data given a base descriptor and a matching pattern. "Version" is completely dependent on the
+ * implementation but is commonly equal to the "last modified" timestamp.
+ *
+ * EXPERIMENTAL: This is implemented explicitly for URIExtractionNamespaceFunctionFactory
```

Comment:
i'd prefer not to have messages like this. When we release QTL, it should be tested in production somewhere.